### ISSUE TITLE
Virtual Keyboard in Gaming Mode makes gamescope crash the deck, possibly connected to bluetooth keyboard issues and de(quertz) layout

### ISSUE THREAD

**Issue by cebe89** [ValveSoftware/gamescope#1571]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using any gamescope patches or a forked version of gamescope?

- [X] The issue occurs on upstream gamescope without any modifications

### Current Behavior

Hello, my deck is reliably crashing whenever I try to enter a text-field in Gaming-mode.

I'm using the most recent stable build (so just the one obtained by updating stable in gaming mode system settings), and the issue appeared a day after I connected a **bluetooth keyboard**.
The keyboard is a "**Seenda KTU-402 Keyboard**". 
I was not able to pair it in gaming mode, but was able to do so in desktop mode, and then returning to gaming mode I could connect to it there too. (this seems to be a known issue with bluetooth, since a few years at least, of which I found many reports here and in the steam forum)
But, it worked fine afterwards, both in gaming as well as in desktop mode, except for some modifications I had to make:
- I had to **restart to get Fn-keys** to work and
- Had to **set my right Alt as the Compose** 3rd level key, as I **am using a german quertz** keyboard layout
- Had to **enforce a de(quertz) layout**, because in gaming mode I still only got a querty even though quertz worked fine in desktop mode. I did this following these modifications: [https://steamcommunity.com/app/1675200/discussions/2/5764050278018994876/?l=german](https://steamcommunity.com/app/1675200/discussions/2/5764050278018994876/?l=german) and [https://github.com/ValveSoftware/SteamOS/issues/798](https://github.com/ValveSoftware/SteamOS/issues/798) which resulted in me adding:
"XKB_DEFAULT_LAYOUT=de
XKB_DEFAULT_VARIANT=quertz" to the "/etc/environment" file (I had to add both actually).

I then recognized that **in gaming mode** every time I **enter a text field**, like in the search bar in the library or the shop, but even ingame, as soon as I pressed a button (either on virtual or bluetooth keyboard), the **deck froze and then crashed**, followed by a restart.

I have tried to unpair the keyboard, but the issue remains now. It **does not appear in desktop mode**.
I have made a systemreport, also uploaded it via the settings system menu in gaming mode, and in desktop mode I also found crashing logs in the "Crashed Processes Viewer" of kde where there are a lot of **gamescope crashes**, which I could confirm to be created once I crash the deck in the gaming mode.

Also now when I connect my bluetooth keyboard in gaming mode (having it paired in desktop mode before obviously), I get a crash just shortly after it has connected successfully, so I am suspecting some **bluetooth driver issue with gamescope** at this point.

Further, when I try to start **gamescope in terminal**, I get a pretty good hint that the **quertz de layout** seems to create the issue, but also when I outcomment the settings in the etc/environment file, the problem remains.

Oh, I have no injectors or anything installed, but I have GE-Proton9-15 and SteamTinkerLaunch v12.12 installed via ProtonUp-Qt, but I don't use them on any games.

### Steps To Reproduce

1. Get a bluetooth keyboard (specifically a "Seenda KTU-402 Keyboard")
2. Pair it in desktop mode
3. Return to gaming mode or restart the deck
4. Turn on Bluetooth, should pair now correctly
5. Return to desktop mode, set compose key to right ALT key in settings menu, add ""XKB_DEFAULT_LAYOUT=de
XKB_DEFAULT_VARIANT=quertz" to the "/etc/environment" file, restart the deck
6. In gaming mode open a text field like in the search bars of library/shop, which results in a crash
7. Restart, go to desktop mode, find crash logs of gamescope in "System/Crashed Processes Viewer"

### Hardware information

```markdown
- Distro: SteamOS 3.5.19 (0240422.1 Build-Id)
- CPU: 8 × AMD Custom APU 0932
- GPU: AMD Radeon Graphics (RADV VANGOGH)
- Driver Version: Mesa 23.3.0-devel (git-4870e346ff)
```

### Software information

```markdown
- Desktop environment: KDE Plasma 5.27.5
- Session type: x11
- Gamescope version: --version doesn't work, so I can't tell, please check my logs
- Gamescope launch command(s): just steamdeck gaming mode, but I also get an error when starting "gamescope" in terminal
```

### Which gamescope backends have the issue you are reporting?

- [X] Wayland (default for nested gamescope)
- [ ] DRM (default for embedded gamescope, i.e. gamescope-session)
- [ ] SDL
- [ ] OpenVR

### Logging, screenshots, or anything else

[systemreport-20241009173047.txt](https://github.com/user-attachments/files/17318117/systemreport-20241009173047.txt)
[gamescope_crashlog.txt](https://github.com/user-attachments/files/17318118/gamescope_crashlog.txt)
[gamescope terminal.txt](https://github.com/user-attachments/files/17318119/gamescope.terminal.txt)

**Comment by cebe89** [comment]
Ah, I found the issue!
Removing the options I set in the etc/environment does remove the issue, but only after a restart.
There is no "quertz" variant for the "de" layout, honestly don't know why I thought there is one, as it clearly isn't shown when showing the possible options. I suppose that I didn't restart the deck after setting them, and as the z button now rightly showed the z key without having to restart, I thought that the variant was needed.

I'll close this one.
Still is related to the other issue I linked regarding the layout set in the settings of the kde doesn't apply to the gaming mode.

**Comment by PietroPizzi69** [comment]
I have set up a SteamMachine and had this problem, but with an existing variant (which I get from "localectl list-x11-keymap-variants"):
'XKB_DEFAULT_LAYOUT=de' works but when I add
'XKB_DEFAULT_VARIANT=de_nodeadkeys' GameMode crashes as I enter something in a textfield, even with controller and onscreen KBD.

I have tested it global in /etc/environment and per user in ~/.config/environment.d/keyboard.conf .

Interestingly, I have no problem with my SteamDeck. At least I hadn't as I last used it with a DE KBD, and I definitely have this settings in the environment file. Maybe and update between then and now changed something.
